botorch/utils/transforms.py

```diff
@@ -166,7 +171,29 @@ def decorated(cls: Any, X: Tensor, **kwargs: Any) -> Any:
                     f" got X with shape {X.shape}."
                 )
             X = X if X.dim() > 2 else X.unsqueeze(0)
-            return method(cls, X, **kwargs)
+            output = method(cls, X, **kwargs)
+            if (
+                assert_output_shape
+                and output.shape != X.shape[:-2]
+                and not (
+                    output.shape == torch.Size() and X.shape[:-2] == torch.Size([1])
+                )  # output of size 1 is typically squeezed
+            ):
+                # the check above may fail when using a batched model.
+                # ensure that this is not the case
+                from botorch.acquisition import AcquisitionFunction
+
+                if (
+                    not isinstance(cls, AcquisitionFunction)
+                    or output.shape != cls.model._input_batch_shape
```

Comment:
this may raise an `AttributeError` for models that don't have an `_input_batch_shape` property.